TST: stats: test_qmc.py::TestPoisson::test_fill_space fails inconconsistently on Accelerate

### Describe your issue.

I'm seeing the test `scipy/stats/tests/test_qmc.py::TestPoisson::test_fill_space[l_bounds1-u_bounds1]` fail in CI here: https://github.com/scipy/scipy/actions/runs/21550611052/job/62098771697?pr=24466

### Reproducing Code Example

```python
# Reproduces problem after about 10 - 200 iterations.
# Must have `pytest-run-parallel` installed to use iterations and forever flag.
spin test -s stats -- -k 'test_qmc and test_fill_space' --iterations=10 -x --forever
```

### Error message

```shell
_______________ TestPoisson.test_fill_space[l_bounds1-u_bounds1] _______________

self = <scipy.stats.tests.test_qmc.TestPoisson object at 0x13a491bf0>
l_bounds = [-5, -5], u_bounds = [5, 5]

    @pytest.mark.parametrize("l_bounds, u_bounds",  # add bounds to test gh-22819
        [(None, None), ([-5, -5], [5, 5]),  ([1.1, 1.1], [5.1, 5.1])])
    def test_fill_space(self, l_bounds, u_bounds):
        radius = 0.2
        engine = self.qmce(d=2, radius=radius, l_bounds=l_bounds, u_bounds=u_bounds)
    
        sample = engine.fill_space()
        # circle packing problem is np complex
>       assert l2_norm(sample) >= radius
E       assert np.float64(0.19999995985843508) >= 0.2
E        +  where np.float64(0.19999995985843508) = l2_norm(array([[-0.77608197,  1.13404075],\n       [-1.01362098,  1.09635316],\n       [-0.78927703,  1.3735868 ],\n       ...,\n       [-4.22197675,  3.72761342],\n       [-4.98296556, -4.5606922 ],\n       [-1.34089559, -4.44003031]], shape=(1530, 2)))

engine     = <scipy.stats._qmc.PoissonDisk object at 0x1515cd5e0>
l_bounds   = [-5, -5]
radius     = 0.2
sample     = array([[-0.77608197,  1.13404075],
       [-1.01362098,  1.09635316],
       [-0.78927703,  1.3735868 ],
       ...,
       [-4.22197675,  3.72761342],
       [-4.98296556, -4.5606922 ],
       [-1.34089559, -4.44003031]], shape=(1530, 2))
self       = <scipy.stats.tests.test_qmc.TestPoisson object at 0x13a491bf0>
u_bounds   = [5, 5]

scipy/stats/tests/test_qmc.py:967: AssertionError
=========================== short test summary info ============================
FAILED scipy/stats/tests/test_qmc.py::TestPoisson::test_fill_space[l_bounds1-u_bounds1] - assert np.float64(0.19999995985843508) >= 0.2
 +  where np.float64(0.19999995985843508) = l2_norm(array([[-0.77608197,  1.13404075],\n       [-1.01362098,  1.09635316],\n       [-0.78927703,  1.3735868 ],\n       ...,\n       [-4.22197675,  3.72761342],\n       [-4.98296556, -4.5606922 ],\n       [-1.34089559, -4.44003031]], shape=(1530, 2)))
= 1 failed, 81878 passed, 7511 skipped, 199 deselected, 307 xfailed, 24 xpassed in 548.06s (0:09:08) =
```

### SciPy/NumPy/Python version and system information

```shell
1.18.0.dev0+git20260131.6e1d64a 2.4.1 sys.version_info(major=3, minor=13, micro=11, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: extraframeworks
    found: true
    has ilp64: false
    include directory: unknown
    lib directory: unknown
    name: Accelerate
    openblas configuration: unknown
    pc file directory: unknown
    version: unknown
  lapack:
    detection method: extraframeworks
    found: true
    has ilp64: false
    include directory: unknown
    lib directory: unknown
    name: Accelerate
    openblas configuration: unknown
    pc file directory: unknown
    version: unknown
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 3.0.1
Compilers:
  c:
    commands: /opt/homebrew/Cellar/llvm/20.1.1/bin/clang
    linker: ld64
    name: clang
    version: 20.1.1
  c++:
    commands: /opt/homebrew/Cellar/llvm/20.1.1/bin/clang++
    linker: ld64
    name: clang
    version: 20.1.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.2.4
  fortran:
    commands: gfortran
    linker: ld64
    name: gcc
    version: 15.1.0
  pythran:
    include directory: ../venv/lib/python3.13/site-packages/pythran
    version: 0.18.1
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
Python Information:
  path: /Users/nick/programming/scipy/venv/bin/python3.13
  version: '3.13'
```

--- a/scipy/stats/tests/test_axis_nan_policy.py
+++ b/scipy/stats/tests/test_axis_nan_policy.py
@@ -494,8 +494,8 @@ def _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
             res = hypotest(*data1d, *args, nan_policy=nan_policy, **kwds)
         res_1db = unpacker(res)
 
-        # changed from 1e-15 solely to appease macosx-x86_64+Accelerate
-        assert_allclose(res_1db, res_1da, rtol=4e-15)
+        # changed from 1e-15 and later from 4e-15 to appease macosx-x86_64+Accelerate
+        assert_allclose(res_1db, res_1da, rtol=1e-14)
         res_1d[i] = res_1db
 
     res_1d = np.moveaxis(res_1d, -1, 0)

--- a/scipy/stats/tests/test_qmc.py
+++ b/scipy/stats/tests/test_qmc.py
@@ -963,8 +963,9 @@ class TestPoisson(QMCEngineTests):
         engine = self.qmce(d=2, radius=radius, l_bounds=l_bounds, u_bounds=u_bounds)
 
         sample = engine.fill_space()
-        # circle packing problem is np complex
-        assert l2_norm(sample) >= radius
+        # rtol to accommodate Accelerate; see gh-24486
+        rtol = 1e-7
+        assert l2_norm(sample) >= radius * (1 - rtol)
 
     def test_bounds_shift_scale(self):
         # test a reasonable property: as long as shape of region is a hypercube,

